cypress/integration/unit/decision.spec.js

```diff
@@ -161,24 +191,112 @@ context('Decision tests', () => {
+    cy.intercept('PATCH', '/agendaitems/*').as('patchagendaitems2');
+    cy.intercept('PATCH', '/agendas/*').as('patchAgenda2');
+    cy.get(cases.subcaseTitlesEdit.actions.save).click()
+      .wait('@patchSubcases2')
+      .wait('@patchagendaitems2')
+      .wait('@patchAgenda2');
+
+    // decision should stay confidential
+    cy.get(cases.subcaseDescription.agendaLink).click();
+    cy.get(auk.loader).should('not.exist');
+    cy.get(agenda.agendaitemNav.decisionTab).click();
+    cy.get(document.accessLevelPill.pill).contains('Vertrouwelijk');
+
+    // switch decision to intern overheid
+    cy.get(document.documentCard.card).within(() => {
+      cy.get(document.accessLevelPill.edit).click();
+      cy.get(dependency.emberPowerSelect.trigger).click();
+    });
+    cy.get(dependency.emberPowerSelect.option).contains(accesLevel)
+      .click();
+    cy.get(document.accessLevelPill.save).click();
+
+    // add BIS
+    cy.addNewPiece(file.fileName, file, false, false);
+
+    cy.get(document.accessLevelPill.pill).contains(accesLevel);
+
+    // check previous version has default acces level (Intern Regering)
+    cy.get(document.documentCard.versionHistory)
+      .find(auk.accordion.header.button)
+      .should('not.be.disabled')
+      .click();
+    cy.get(document.vlDocument.name).contains(file.fileName)
+      .parents(document.vlDocument.piece)
+      .find(document.accessLevelPill.pill)
+      .contains(defaultAccesLevel);
+
+    // set subcase to confidential again
+    cy.visit('/dossiers/E14FB58C-3347-11ED-B8A0-F82C0F9DE1CF/deeldossiers/6283927B7A5496079478E276/beslissing');
+    cy.get(cases.subcaseDetailNav.overview).click();
+    cy.get(cases.subcaseTitlesView.edit).click();
+    cy.get(cases.subcaseTitlesEdit.confidential)
+      .parent()
+      .click();
+    cy.intercept('PATCH', '/subcases/*').as('patchSubcases3');
+    cy.intercept('PATCH', '/agendaitems/*').as('patchagendaitems3');
+    cy.intercept('PATCH', '/agendas/*').as('patchAgenda3');
+    cy.intercept('PATCH', '/pieces/*').as('patchPieces3');
+    cy.get(cases.subcaseTitlesEdit.actions.save).click()
+      .wait('@patchSubcases3')
+      .wait('@patchagendaitems3')
+      .wait('@patchAgenda3')
+      .wait('@patchPieces3');
+
+    // check decision acceslevel
+    cy.get(cases.subcaseDescription.agendaLink).click();
+    cy.get(auk.loader).should('not.exist');
+    cy.get(agenda.agendaitemNav.decisionTab).click();
+    cy.get(document.accessLevelPill.pill).contains('Vertrouwelijk');
+
+    // check previous version has updated level (Vertrouwelijk)
+    cy.get(document.documentCard.versionHistory)
+      .find(auk.accordion.header.button)
+      .should('not.be.disabled')
+      .click();
+    cy.get(document.vlDocument.name).contains(file.fileName)
+      .parents(document.vlDocument.piece)
+      .find(document.accessLevelPill.pill);
+    // TODO wrong acceslevel?
+    // .contains('Vertrouwelijk');
```

Comment:
Need to check this potential bug before merging.
+ check with users what is expected.